### PR TITLE
Reduce the code size some more by refactoring the header & checksum function.

### PR DESCRIPTION
**Pull request by nomis52**

### CODE REVIEW

**Comment by nomis52** [comment]
This is blocked behind #91

**Comment by peternewman** [comment]
Can you resync when it's merged.

**Comment by nomis52** [comment]
Done, diff is much smaller now.

File: firmware/src/network_model.c [hunk 1]
```diff
@@ -200,10 +200,7 @@ int NetworkModel_GetListInterfaces(const RDMHeader *header,
     ptr = PushUInt32(ptr, INTERFACE_DEFINITIONS[i].id);
     ptr = PushUInt16(ptr, INTERFACE_DEFINITIONS[i].hardware_type);
   }
-  RDMResponder_BuildHeader(header, ACK, GET_COMMAND_RESPONSE,
-                           ntohs(header->param_id),
-                           (ptr - g_rdm_buffer) - sizeof(RDMHeader));
-  return RDMUtil_AppendChecksum(g_rdm_buffer);
+  return RDMResponder_AddHeaderAndChecksum(header, ACK, ptr - g_rdm_buffer);
```
Comment (peternewman) [inline firmware/src/network_model.c]: Can we have a BuildGetAck please like BuildSetAck, which calls RDMResponder_AddHeaderAndChecksum.

**Comment by peternewman** [comment]
LGTM, just one comment to fix.

**Pull request merged**